src/main/java/bio/terra/service/snapshot/SnapshotService.java

```diff
@@ -727,116 +742,25 @@ public AssetSpecification buildAssetFromSnapshotAccessRequest(
   }
 
   @VisibleForTesting
-  List<SnapshotBuilderTable> pullTables(SnapshotAccessRequestResponse snapshotRequestModel) {
-    var valueSets = snapshotRequestModel.getSnapshotSpecification().getValueSets();
-    var valueSetNames = valueSets.stream().map(SnapshotBuilderFeatureValueGroup::getName).toList();
+  List<SnapshotBuilderTable> pullTables(
+      SnapshotAccessRequestResponse snapshotAccessRequest, SnapshotBuilderSettings settings) {
+    List<String> includedTableNames =
+        snapshotAccessRequest.getSnapshotSpecification().getOutputTables().stream()
+            .map(SnapshotBuilderOutputTable::getName)
+            .toList();
 
-    Map<String, SnapshotBuilderTable> tableMap = populateManualTableMap();
+    List<SnapshotBuilderDatasetConceptSet> allTables = settings.getDatasetConceptSets();
 
-    Set<String> missing = new HashSet<>(valueSetNames);
-    missing.removeAll(tableMap.keySet());
+    Set<String> missing = new HashSet<>(includedTableNames);
+    allTables.stream().map(SnapshotBuilderDatasetConceptSet::getName).forEach(missing::remove);
     if (!missing.isEmpty()) {
       throw new IllegalArgumentException("Unknown value set names: " + missing);
```

Comment:
Nit: What do you think about changing this error message now that we don't have value sets?

Reply:
yeah, maybe unknown concept set name?